external/mbedtls/alt/pk_wrap_alt.c

```diff
@@ -870,6 +964,8 @@ int eckey_sign_wrap(void *ctx, mbedtls_md_type_t md_alg,
 		mbedtls_ecdsa_free(&ecdsa);
 		return MBEDTLS_ERR_ECP_INVALID_KEY;
 	}
+	key_idx = (((mbedtls_ecp_keypair *) ctx)->grp).key_index;
+
```

Comment:
I'm so sorry, but would you please move this to line 1045?
(for the code unifomity with 4.0)

`.alt` codes are slightly different in 3.1 and 4.0, but 4.0 is the latest code.
So it would be more efficient to sync with 4.0.

```
/*
	 * 2. Sign
	 */
	if ((ret = sl_init(&shnd)) != SECLINK_OK) {
		if (!is_injected_key) {
			free(prikey.data);
		}
		free(t_hash.data);
		return MBEDTLS_ERR_ECP_HW_ACCEL_FAILED;
	}
	if (!is_injected_key) {
		key_idx = alt_set_key(shnd, key_type, NULL, &prikey, 32);
		free(prikey.data);
		if (key_idx < 0) {
			free(t_hash.data);
			sl_deinit(shnd);
			return MBEDTLS_ERR_ECP_HW_ACCEL_FAILED;
		}
	}
	key_idx = (((mbedtls_ecp_keypair *) ctx)->grp).key_index;
	ret = sl_ecdsa_sign_md(shnd, ecdsa_mode, &t_hash, key_idx, &sign);
	if (!is_injected_key) {
		sl_remove_key(shnd, key_type, key_idx);
	}
	sl_deinit(shnd);
	free(t_hash.data);
```

Reply:
updated